math/utils.go

```diff
@@ -94,7 +95,7 @@ func StdDev(data []Dec) (Dec, error) {
 	// Calculate the mean, excluding NaN values
 	for _, v := range data {
 		if v.isNaN { // Check if the value is NaN
-			continue // Skip NaN values
+			return Dec{}, errors.New("input data contains NaN values")
```

Comment:
my other PR introduces earlier validation on these functions right on the start of the function, for all functions in this file.

Reply:
great, pulled your changes into my PR and merged them

Reply:
ty. 
I also see that in loop cases like this, your code is more efficient, so feel free to keep it.